### ISSUE TITLE
Add metrics to keep track of the total time the devices are connected to Hono

### ISSUE THREAD

**Issue by kaniyan** [eclipse/hono#1287]
Add metrics to keep track of the total time the devices are connected to Hono per tenant. 

Currently there are metrics in Hono to track the number of simultaneous device connections. In addition to it, it is also valuable to track how long the devices are connected to Hono per tenant.   This can also be used to effectively manage the resources by setting the _resource-limits_ and also shown in the dashboard.

**Comment by kaniyan** [comment]
The MQTT and AMQP protocol adapters which maintain connection state already provide metrics namely `hono.connections` to track the number of connected devices per tenant.

Now we would also like to find out the total time the devices are connected per tenant. After taking a closer look at the existing metrics, I think the metrics `hono.connections` could well also be used to compute the total connection time per tenant. Currently this metric is being recorded using type _Gauge_, scrapped at regular intervals and stored as time series in the Prometheus server. By finding the sum of the number of device connections reported at every scrape for the given time period and multiplying it with the scrape interval should give the total device connection time per tenant. 

A sample PromQL query to get the total connection time of a tenant for the last one hour is given below:

```
sum(sum_over_time(hono_connections_authenticated {tenant="DEFAULT_TENANT"} [1h])) * scraping interval

```

The applications like resource limit checks that would like to calculate the total connection time should also be aware of the scrapping interval. 

In this approach, the level of accuracy depends on the scrapping interval. For example if the scrapping interval is 5 seconds, then a connection established and disconnected within 5 seconds may not be recorded by the metrics `hono.connections` and so the connection time.

**Comment by sophokles73** [comment]
I am a little concerned with the fact that the usefulness of this approach is so closely tied to the definition of an appropriate scraping interval. When using a short interval, the accurateness of the measures values might be sufficient. However, I am not sure if 5 seconds is a reasonable interval length in all cases.
Using a longer interval might in fact lead to a connection time of 0. For example when using a 20 second scrape interval, if all devices connect, send data and disconnect every 20 seconds, we might end up in a situation where the devices always connect in between the scrapes, right? FMPOV this example isn't very far fetched. Even if only, say, half of the devices connect in between the scrapes, the resulting accuracy of the measurement won't be sufficient, would it?

**Comment by kaniyan** [comment]
yes as the scraping interval gets longer and quicker the devices connect and disconnect, then less accurate the computed connection time using the metrics `hono.connections`. I think then it makes sense to make use of a new metrics to record the connection time.

**Comment by sophokles73** [comment]
> I think then it makes sense to make use of a new metrics to record the connection time.

I think so too

**Comment by kaniyan** [comment]
A straightforward solution that comes to my mind is to start a timer immediately after a device successfully connects, stop that timer once that device disconnects and record the duration using metrics of type _Summary_ or _Counter_. After giving some thought, I think this way is not very useful as the data will be stored in Prometheus only after a device disconnects. In case of MQTT or AMQP, it is possible that the devices stay connected for a long time like weeks, months or even longer. If _resource-limit_ checks are to make use of this data to set a limit based on the device connection time, then this data won't be accurate and may not be quite useful.

As an alternative, how about using a timer per _tenant_ that records periodically lets say every 10 seconds the connection time of all devices under this tenant (i.e no. of devices connected at the start of the timer multiplied by the timer interval will be recorded using a metric type of `Counter`)? There will be devices that connect or disconnect after the timer is started. I think this can be handled by stopping that timer, recording the connection time and then start the timer to track the connection time based on the change in the number of connected devices. WDYT? Ofcourse other suggestions are surely welcome :)

**Comment by sophokles73** [comment]
@kaniyan can this be closed?

**Comment by kaniyan** [comment]
> @kaniyan can this be closed?

yes. This has been closed via PR #1678.
